crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,166 @@
+/// Options to control the behavior of [`spawn`].
+///
+/// Refer to the field-level documentation for more information about each individual options.
+///
+/// The defaults are ok for most use cases: `SpawnOptions::default()`.
+/// Use the builder pattern to customize them further:
+/// ```no_run
+/// let opts = re_sdk::SpawnOptions::default().with_port(1234u16).with_memory_limit("25%");
+/// ```
+#[derive(Debug, Clone, Default)]
+pub struct SpawnOptions {
+    /// The port to listen on.
+    ///
+    /// Defaults to `9876` if unspecified.
+    pub port: Option<u16>,
+
+    /// An upper limit on how much memory the Rerun Viewer should use.
+    /// When this limit is reached, Rerun will drop the oldest data.
+    /// Example: `16GB` or `50%` (of system total).
+    ///
+    /// Defaults to `75%` if unspecified.
+    pub memory_limit: Option<String>,
```

Comment:
We could generalize this to a list of arguments instead, with a helper builder-method for setting `--memory-limit`

Reply:
I didnt understand your comment

Reply:
```rs
/// Extra arguments passed to the `rerun` process
pub extra_args: Vec<string>
```

```rs
impl SpawnOptions {
    pub fn with_memory_limit(self, memory_limit: String) {
        self.extra_args.push(format!("--memory-limit={memory_limit}));
        self
    }
}
```

Reply:
I don't see the upside of this vs. what we have right now with strong types, docs, ease of introspection etc.

Reply:
The upside is the user can set ALL arguments that the `rerun` binary supports, like `--drop-at-latency`, `--profile`, `--window-size` etc without us having to repeat them here again, though we can still have helpers (`with_memory_limit`) for the common ones to get that type safety

Reply:
I've added a catch all `extra_args`.